lib/compat/wordpress-6.0/class-wp-theme-json-gutenberg.php

```diff
@@ -161,4 +161,104 @@ public function get_data() {
 		return $flattened_theme_json;
 	}
 
+	/**
+	 * Merge new incoming data.
+	 *
+	 * @since 5.8.0
+	 * @since 5.9.0 Duotone preset also has origins.
+	 * @since 6.0.0 Calls functional_merge internally
+	 *
+	 * @param WP_Theme_JSON $incoming Data to merge.
+	 */
+	public function merge( $incoming ) {
+		$this->theme_json = static::functional_merge( $this->theme_json, $incoming->get_raw_data() );
+	}
+
+	/**
+	 * Merge two theme.json data structures
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param array $base Data to merge onto.
+	 * @param array $replacements Data to add onto $base
+	 * @return array The result of the merge
+	 */
+	public static function functional_merge( $base, $replacements ) {
```

Comment:
The idea of this function is that it will work with any 2 arrays and do the same thing, so we can use the same logic without being in context of an object.